### ISSUE TITLE
t.conch.manhole does not exit continuation prompt after ctrl-c

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2663]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| PenguinOfDoom reported|
|-|-|
|Trac ID|trac#2663|
|Type|defect|
|Created|2007-05-18 23:57:02Z|

Consider manhole session:
```
>>> (
... 
KeyboardInterrupt
... 
```

Compared to real Python interpreter:

```
>>> (
... 
KeyboardInterrupt
>>> 
```

<details><summary>Searchable metadata</summary>

```
trac-id__2663 2663
type__defect defect
reporter__PenguinOfDoom PenguinOfDoom
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1179532622000000 1179532622000000
changetime__1179700978000000 1179700978000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

fix wiki formatting

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Fixed in interrupt-during-continuation-2663

**Comment by twisted-trac** [comment]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|PenguinOfDoom set owner to @exarkun|
|-|-|

Looks fine.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [20343]) Merge interrupt-during-continuation-2663

Author: exarkun
Reviewer: PenguinOfDoom
Fixes [#2663](https://github.com/twisted/twisted/issues/2663)

Add tests for ^C during an input line continuation in manhole and fix
the implementation to update the prompt and the line buffer to correctly
reflect the consequences (ie, clear them).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
